gigadb/app/tools/wasabi-migration/generate_config.sh

```diff
@@ -0,0 +1,74 @@
+    source "./.env"
+fi
+
+# Fetch and set environment variables from GitLab
+if ! [ -s ./.secrets ];then
+    # deal with special case we are in Upstream
+    if [[ $CI_PROJECT_URL == "https://gitlab.com/gigascience/upstream/gigadb-website" ]];then
+      PROJECT_VARIABLES_URL="https://gitlab.com/api/v4/projects/gigascience%2Fupstream%2Fgigadb-website/variables"
+    fi
+
+    echo "Retrieving variables from ${GROUP_VARIABLES_URL}"
+    curl -s --header "PRIVATE-TOKEN: $GITLAB_PRIVATE_TOKEN" "${GROUP_VARIABLES_URL}" | jq -r '.[] | select(.key != "ANALYTICS_PRIVATE_KEY") | .key + "=" + .value' > .group_var
+
+    if [[ $CI_PROJECT_URL != "https://gitlab.com/gigascience/upstream/gigadb-website" ]];then
+      echo "Retrieving variables from ${FORK_VARIABLES_URL}"
+      curl -s --header "PRIVATE-TOKEN: $GITLAB_PRIVATE_TOKEN" "${FORK_VARIABLES_URL}?per_page=100" | jq -r '.[] | select(.key != "ANALYTICS_PRIVATE_KEY") | .key + "=" + .value' > .fork_var
+    fi
+
+    echo "Retrieving variables from ${PROJECT_VARIABLES_URL}"
+    curl -s --header "PRIVATE-TOKEN: $GITLAB_PRIVATE_TOKEN" "${PROJECT_VARIABLES_URL}?per_page=100&page=1"  > .project_var_raw1
+    curl -s --header "PRIVATE-TOKEN: $GITLAB_PRIVATE_TOKEN" "${PROJECT_VARIABLES_URL}?per_page=100&page=2"  > .project_var_raw2
+    jq -s 'add' .project_var_raw1 .project_var_raw2 > .project_vars.json
+    cat .project_vars.json | jq --arg ENVIRONMENT $GIGADB_ENV -r '.[] | select(.environment_scope == "*" or .environment_scope == "dev" ) | select(.key | test("private_key|tlsauth|ca|pem|cert";"i") | not ) |.key + "=" + .value' > .project_var
+
+    echo "Retrieving variables from ${MISC_VARIABLES_URL}"
+    curl -s --header "PRIVATE-TOKEN: $GITLAB_PRIVATE_TOKEN" "${MISC_VARIABLES_URL}?per_page=100" | jq --arg ENVIRONMENT $GIGADB_ENV -r '.[] | select(.environment_scope == "*" or .environment_scope == $ENVIRONMENT ) | select(.key | test("sftp_") ) | .key + "=" + .value' > .misc_var
+
+    # Need to account for the fact that there is no .fork_var when dealing with
+    # upstream configuration. An error will be generated if we try to cat a
+    # non-existent file
+    if [ "$CI_PROJECT_URL" == "https://gitlab.com/gigascience/upstream/gigadb-website" ];
+    then
+      cat .group_var .project_var .misc_var > .secrets && rm .group_var && rm .project_var && rm .misc_var && rm .project_var_raw1 && rm .project_var_raw2 && rm .project_vars.json
+    else
+      # Fork configuration
+      cat .group_var .fork_var .project_var .misc_var > .secrets && rm .group_var && rm .fork_var && rm .project_var && rm .misc_var && rm .project_var_raw1 && rm .project_var_raw2 && rm .project_vars.json
+    fi
+
+    echo "# Some help about this file in ops/configuration/variables/secrets-sample" >> .secrets
+fi
+echo "Sourcing secrets"
+source "./.secrets"
+
+# Restore default settings for variables
+set +a
+
+# Generate rclone configuration
+SOURCE=${APP_SOURCE}/rclone.conf.dist
+TARGET=${APP_SOURCE}/config/rclone.conf
```

Comment:
Hi @pli888,

`${APP_SOURCE}/config/` is not present, when `% docker-compose run --rm config` would get this error:
```
+ SOURCE=/app/rclone.conf.dist
+ TARGET=/app/config/rclone.conf
+ VARS='$WASABI_ACCESS_KEY_ID:$WASABI_SECRET_ACCESS_KEY'
+ envsubst '$WASABI_ACCESS_KEY_ID:$WASABI_SECRET_ACCESS_KEY'
/app/generate_config.sh: line 74: /app/config/rclone.conf: No such file or directory
ERROR: 1
```
If `mkdir config` and then `docker-compose run --rm config`, `rclone.conf` would then be generated.